cmd/kubeadm/app/util/pkiutil/pki_helpers.go

```diff
@@ -59,6 +61,27 @@ const (
 	rsaKeySize             = 2048
 )
 
+var (
+	reusableSigner   crypto.Signer
+	reuseSigner      bool
+	reuseSignerMutex = &sync.Mutex{}
+)
+
+func init() {
+	// Reuse keys during unit tests by default
+	if strings.HasSuffix(os.Args[0], ".test") {
+		SetReuseSigner(true)
```

Comment:
using os.Args is sketchy but i can't find another reliable way.

Reply:
yeah... this is not something I would recommend. Instead, I'd do something like this:

1. Make `NewPrivateKey` able to be overridden
2. Make a test-only package that swaps out an implementation of NewPrivateKey that pulls private keys from a testdata folder
3. In packages that are slow due to private key generation, install that test helper in a `TestMain` function.

This guarantees that the behavior change only affects tests, and also speeds up all private key generations (not just CA generation)

opened https://github.com/kubernetes/kubernetes/pull/98664 with this as a proposal

Reply:
ok, given https://github.com/kubernetes/kubernetes/pull/98664 i will exclude the PKI changes from this PR and only leave the "dynamic defaults".